mirgecom/eos.py

```diff
@@ -345,48 +536,96 @@ def internal_energy(self, cv: ConservedVars):
         .. math::
 
             e = \rho{E} - \frac{1}{2\rho}(\rho\vec{V} \cdot \rho\vec{V})
+
+        Parameters
+        ----------
+        cv: :class:`~mirgecom.fluid.ConservedVars`
+            :class:`~mirgecom.fluid.ConservedVars` containing at least the mass
+            ($\rho$), energy ($\rho{E}$), momentum ($\rho\vec{V}$), and the vector
+            of species masses, ($\rho{Y}_\alpha$).
+
+        Returns
+        -------
+        :class:`~meshmode.dof_array.DOFArray`
+            Internal energy of the fluid
         """
         return (cv.energy - self.kinetic_energy(cv))
 
-    def get_density(self, pressure, temperature, species_fractions):
+    def get_density(self, pressure, temperature, species_mass_fractions):
         r"""Get the density from pressure, temperature, and species fractions (Y).
 
         The gas density $\rho$ is calculated from pressure, temperature and $R$ as:
 
         .. math::
 
             \rho = \frac{p}{R_s T}
+
+        Parameters
+        ----------
+        pressure: :class:`~meshmode.dof_array.DOFArray`
+            The fluid pressure
+        temperature: :class:`~meshmode.dof_array.DOFArray`
+            The fluid temperature
+        species_mass_fractions: numpy.ndarray
+            Object array of species mass fractions
+
+        Returns
+        -------
+        :class:`~meshmode.dof_array.DOFArray`
+            The total fluid mass density
         """
         return self._pyrometheus_mech.get_density(pressure, temperature,
-                                                  species_fractions)
+                                                  species_mass_fractions)
 
-    def get_internal_energy(self, temperature, species_fractions):
+    def get_internal_energy(self, temperature, mass=None,
+                            species_mass_fractions=None):
         r"""Get the gas thermal energy from temperature, and species fractions (Y).
 
         The gas internal energy $e$ is calculated from:
 
         .. math::
 
             e = R_s T \sum{Y_\alpha h_\alpha}
+
+        Parameters
+        ----------
+        temperature: :class:`~meshmode.dof_array.DOFArray`
+            The fluid temperature
+        species_mass_fractions: numpy.ndarray
+            Object array of species mass fractions
+        mass:
+            Unused
         """
+        if species_mass_fractions is None:
+            raise ValueError("Mixture EOS.get_internal_energy requires "
+                             "species_mass_fractions argument.")
```

Comment:
Can do the `=None` removal here too I think.

Reply:
(82cbd73)